httpc.c

```diff
@@ -1395,12 +1396,22 @@ static int httpc_op_heap(httpc_options_t *a, const char *url, int op, httpc_call
 			return HTTPC_ERROR;
 		memset(h, 0, sizeof *h);
 		a->state     = h;
-		h->os        = a;
-		h->rcv       = rcv;
-		h->snd       = snd;
-		h->rcv_param = rcv_param;
-		h->snd_param = snd_param;
 	}
+	else if(!httpc_is_yield_on(h)) {
+		/* Clear states when re-using non-yielding sockets */
```

Comment:
This will cause a problem if you are reusing the connection and yielding I think. I'd remove this, and reset these parameters if `r = HTTPC_OK` after `httpc_state_machine`. (I think that would work).

Reply:
Ah, that's much cleaner. Thanks!